v2/dtos/requests/deviceprofile.go

```diff
@@ -0,0 +1,212 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package requests
+
+import (
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v2"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+// AddDeviceProfileRequest defines the Request Content for POST DeviceProfile DTO.
+// This object and its properties correspond to the AddDeviceProfileRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/AddDeviceProfileRequest
+type AddDeviceProfileRequest struct {
+	common.BaseRequest `json:",inline"`
+	Name               string                 `json:"name" yaml:"name" validate:"required" `
+	Manufacturer       string                 `json:"manufacturer,omitempty" yaml:"manufacturer,omitempty"`
+	Description        string                 `json:"description,omitempty" yaml:"description,omitempty"`
+	Model              string                 `json:"model,omitempty" yaml:"model,omitempty"`
+	Labels             []string               `json:"labels,omitempty" yaml:"labels,flow,omitempty"`
+	DeviceResources    []dtos.DeviceResource  `json:"deviceResources,omitempty" yaml:"deviceResources,omitempty" validate:"dive"`
```

Comment:
`DeviceResources` also need `required`? Looks like `dive` only applied if `DeviceResources` provided. Also remove `omitempty's` if at least one device resource is required.